torch_geometric/nn/models/explainer_settings.py

```diff
@@ -0,0 +1,133 @@
+from typing import Tuple
+
+import torch
+
+from torch_geometric.nn.models.explainer_algorithms import ExplainerAlgorithm
+
+
+class ExplainerSetter(torch.nn.Module):
+    r"""A user configuration class for the instance level-explanation
+    of Graph Neural Networks, *e.g.* :class:`~torch_geometric.nn.GCN`.
+
+    Args:
+        explanation_type (str): the type of explanation to be computed.
```

Comment:
this might be related to objectives. e.g. for `model`, it makes sense to use MI as objective; for `phenomenon`, we might use the logit corresponding to the ground truth class.
So when we construct say `Saliency(ExplainerAlgorithm)`, we can take in this and create the objective accordingly.